automation/base/driver/BaseDriver.ts

```diff
@@ -92,7 +94,23 @@ export default class BaseDriver {
         return this.Page.getByTestId(testId);
     }
 
-    public getPage<T>(type: { new(page: Page): T; }) {
+    public getPage<T>(type: { new (page: Page): T }) {
         return new type(BaseDriver.focusedDriver.Page);
     }
-}
\ No newline at end of file
+
+    public async executeFunc(func: any, attepmts: number) {
+        let message = "";
+        for (let i = 0; i < attepmts; i++) {
+            try {
+                await func();
+                return;
+            } catch (err) {
+                console.log(`${i + 1} attempt to execute ${func.name}`);
+                message = err;
+            }
+        }
+        if (!(message === "")) {
+            throw new Error(message);
```

Comment:
Better to rethrow the actual Error object rather than just the message string. It gives more information about the error.